### ISSUE TITLE
SearchServiceExtension->generateSearchUUID is misnamed

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-search-service#16]
A UUID is a specific concept (or spec) in computer science terms, and this isn't what we're creating here. See https://en.wikipedia.org/wiki/Universally_unique_identifier.

I'd suggest `generateSearchDocumentID()`. Or I would question why this needs to be extended onto DataObject in the first place, since it deals with public properties on DataObject only. Couldn't we just keep this in `DataObjectDocument->getIdentifier()`? 

I don't see that logic being customised often. Even for subsites, each page has a unique database identifier. It's only an issue for fluent... is that an extension of `DataObjectDocument`? Alternatively, we require 4.6 and use `DataObject->getUniqueID()` which has just been merged. That handles special cases like fluent.

**Comment by unclecheese** [comment]
Yup, i think this has been superseded by the `DocumentInterface` abstraction, so let's leave it to that implementation.

**Comment by unclecheese** [comment]
https://github.com/silverstripe/silverstripe-search-service/commit/6a62ddaec57034b8ee33e49f07d5adbe076b50d6#diff-50b6fa13029e606508e95c84bbe77a19R71

**Comment by chillu** [comment]
So you didn't want to require 4.6 or newer and go with `getUniqueID()`? That'll give you fluent compat out of the box for example (on identifier generation at least).

**Comment by chillu** [comment]
Moving this back to "in progress", got an outstanding question - `DataObject->getUniqueKey()`

**Comment by michalkleiner** [comment]
From my perspective, it makes sense to use `getUniqueID()`.
CMS is on 4.9 now and this module doesn't have a stable release yet so should be fine.
